lib/tty/prompt/multi_list.rb

```diff
@@ -23,6 +23,8 @@ def initialize(prompt, **options)
         @echo = options.fetch(:echo, true)
         @min  = options[:min]
         @max  = options[:max]
+        error = "min must not be greater than max"
+        raise ConfigurationError, error if @min && @max && @min > @max
```

Comment:
I'd extract to a separate private method like `check_minmax` or similar like so:

```ruby
# @api private
def check_minmax
  return unless @min && @max && @min > @max

  raise ConfigurationError, "min must not be greater than max"
end
```

Then you can reuse this method for the `min` and `max` methods right after the initializer.

Reply:
I took this to mean that instead of invoking this method from the initializer that we should invoke it from both the `min` and the `max` setter methods. I wasn't sure exacly where in the file you'd like this method defined; I opted for in the `private` section, close to the `minmax_help` method.